src/local-peers.js

```diff
@@ -59,16 +60,23 @@ class Peer {
   #disconnectedAt = 0
   /** @type {Protomux<import('@hyperswarm/secret-stream')>} */
   #protomux
+  #log
 
   /**
    * @param {object} options
    * @param {Buffer} options.publicKey
    * @param {ReturnType<typeof Protomux.prototype.createChannel>} options.channel
+   * @param {Logger} [options.logger]
    */
-  constructor({ publicKey, channel }) {
+  constructor({ publicKey, channel, logger }) {
     this.#publicKey = publicKey
     this.#channel = channel
     this.#connected = pDefer()
+    // @ts-ignore
+    this.#log = (formatter, ...args) => {
+      const log = Logger.create('peer', logger).log
+      return log.apply(null, [`[%h] ${formatter}`, publicKey, ...args])
+    }
```

Comment:
> what exactly is this ts ignoring here? the arguments not having explicit type annotations?

Yes, it's ignoring not having type annotations, because even adding `any` is complicated for this I think, and didn't seem worth the effort.

> this is basically enabling the usage of the same logger for all peer instances, but adding the appropriate peer-specific info to the log message?

Yes, it's providing a helper function that includes the peerId in each log message, to simplify the logging in the class. When reading these logs we need to know which peer each log message refers to.